[TOOLS-4704] when migration using SQL, the target schema enters null pt.2

Kang-dot opened the pull request:
http://jira.cubrid.org/browse/TOOLS-4704

- Purpose
SQL을 이용한 이관 기능 사용 시 schema 정보가 누락되어 쿼리가 정확하게 수행이 안되는 문제를 수정합니다

- Implementation
target이 cubrid dump가 되는 경우 combo box가 제대로 적용되도록 수정했습니다

- Remarks
아래 이슈의 후속 수정입니다
https://github.com/CUBRID/cubrid-migration/pull/268

Kang-dot committed:
[TOOLS-4704] apply SQL migration function to cubrid dump

Kang-dot committed:
[TOOLS-4704] resolve when source owner is null issue

- set source owner to source connection owner

Kang-dot committed:
[TOOLS-4704] format files

- format edited files

Kang-dot committed:
[TOOLS-4704] fix ui when target has no user schema

Kang-dot committed:
[TOOLS-4704] remove final string

- remove noticie string. it is not necessary

Kang-dot committed:
[TOOLS-4704] remove unncessary code

- remove unncessary code
- apply format

Kang-dot committed:
[TOOLS-4704] change method structure

- change method structure
- add format

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/core/engine/importer/impl/LoadFileImporter.java:
```diff
@@ -176,14 +176,15 @@ protected void handleDataFile(
             MigrationDirAndFilesManager mdfm = mrManager.getDirAndFilesMgr();
             String schemaName =
                     config.getSrcCatalog().getDatabaseType().isSupportMultiSchema()
+                                    && !(stc.getOwner() == null)
```
`!=` 을 사용하는 것이 좋아보입니다.

```suggestion
                                    && stc.getOwner() != null
```

Kang-dot replied:
확인했습니다 내용을 수정합니다

Srltas commented on com.cubrid.cubridmigration.core/src/com/cubrid/cubridmigration/core/engine/config/MigrationConfiguration.java:
```diff
@@ -4989,14 +4996,11 @@ public void setSrcCatalog(Catalog srcCatalog, boolean reset) {
     public void setTarCatalog(Catalog tarCatalog) {
-        if (tarCatalog == null) {
-            throw new IllegalArgumentException("Target Catalog cannot be null");
-        }
```
target catalog가 null이 될 수 있도록 변경된 상황이라면, 해당 값을 반환하는 메서드(getTarCatalog 등)도 Optional<Catalog>을 반환하도록 수정하는 방향은 어떨까요?

현재 getTarCatalog()를 사용하는 곳은 SQLTableManageView 단 한 곳으로 보이므로, 변경 범위도 제한적일 것으로 생각됩니다.

주석으로 null 가능성을 명시하는 것도 방법이지만, Optional을 통해 null 처리를 컴파일 타임에 강제하는 쪽이 더 안전하고 의도를 명확하게 전달할 수 있다고 생각합니다.

Kang-dot replied:
확인했습니다 targetCatalog에 대한 getter에서 optional을 반환하도록 변경했고 getter를 호출하는 SQLTableManageView의 메소드도 수정되었습니다

Srltas commented on com.cubrid.cubridmigration.ui/src/com/cubrid/cubridmigration/ui/wizard/page/view/SQLTableManageView.java:
```diff
@@ -610,15 +611,22 @@ public void run(IProgressMonitor monitor)
-                        .toArray(String[]::new);
-        this.tarSchemaList = schemaNameArr;
+        if (config.isAddUserSchema()) {
+            Catalog catalog = (config.getTarCatalog() != null)
```
소괄호가 꼭 필요해 보이지는 않습니다.
```suggestion
            Catalog catalog = config.getTarCatalog() != null
```

Kang-dot replied:
확인했습니다 다만 이 부분은 optional을 적용하면서 코드가 전체적으로 변경되었습니다

Kang-dot committed:
[TOOLS-4704] apply changes

- use Optional in MigrationConfiguration's getTarCatalog method
- edit wrong code

Kang-dot committed:
[TOOLS-4704] apply format

- solve github-checks/code style failure

Kang-dot committed:
[TOOLS-4704] apply locale in toUpperCase

- apply Locale.US in toUpperCase methods

Kang-dot committed:
[TOOLS-4704] apply format

- apply format
